samples/cpp/gstreamer_pipeline.cpp

```diff
@@ -0,0 +1,207 @@
+        ! vaapidecodebin  disable-vpp=true            \
+        ! deinterlace                                 \
+        ! autovideoconvert                            \
+        ! videoconvert                                \
+            dither=0                                  \
+            dither-quantization=0                     \
+            chroma-resampler=0                        \
+            alpha-mode=1                              \
+            alpha-value=0                             \
+            chroma-mode=3                             \
+        ! appsink sync=false";
+        cap = VideoCapture(pipeline, CAP_GSTREAMER);
+    }
+    else if (arg_plugin == "gst-libav")
+    {
+        // Choose a codec for libav
+        if (arg_container == "mpeg4")
+        {
+            codec_name = "avdec_mpeg4";
+        }
+        else if (arg_container == "h264")
+        {
+            container_plugin = "avdec_h264";
+        }
+        else if (arg_container == "h265")
+        {
+            container_plugin = "avdec_h265";
+        }
+        else
+        {
+            help();
+            cout << "Not find codec" << endl;
+            return -3;
+        }
+
+        string pipeline = "filesrc location=\"" + arg_file_name + "\" \
+        ! " + container_plugin + "                    \
+        ! " + codec_name + "                          \
+        ! videoconvert                                \
+        ! appsink sync=false";
+        cap = VideoCapture(pipeline, CAP_GSTREAMER);
+    }
+    else if (arg_plugin == "ffmpeg")
+    {
+        cap = VideoCapture(arg_file_name, CAP_FFMPEG);
+    }
+    else
+    {
+        help();
+        cout << "Not find GStreamer pipeline" << endl;
+        return -4;
+    }
+
+    // Choose a show video or mesuare fps
+    if (arg_start == "show_video")
+    {
+        namedWindow("Display window", WINDOW_AUTOSIZE);
+        while(true)
+        {
+            Mat frame;
+            cap >> frame;
+            if (frame.empty()) { break; }
+            imshow("Display window", frame);
+            // press ESC
+            if (waitKey(1) == 27) { break; }
+        }
+        destroyAllWindows();
+    }
+    else if (arg_start == "mesuare_fps")
+    {
+        cout << "_____________________________________" << '\n';
+        cout << "Start measure frame per seconds (fps)" << '\n';
+        cout << "Loading ..." << '\n';
+
+        Mat frame;
+        unsigned count_frame = 0;
+        int64 result_time = 0;
+        vector<int64> tick_counts;
+        double time_fps = 0.0;
+
+        while(true)
+        {
+            int64 temp_count_tick = getTickCount();
+            cap >> frame;
+            temp_count_tick = getTickCount() - temp_count_tick;
+            if (frame.empty()) { break; }
+            tick_counts.push_back(temp_count_tick);
+            count_frame++;
+        }
+
+        for (size_t i = 0; i < tick_counts.size(); i++)
```

Comment:
Can the following calculations be replaced with `cv::mean`, `cv::sum`, `vector::size` calls?